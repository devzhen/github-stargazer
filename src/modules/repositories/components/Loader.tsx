import React from 'react';
import { ActivityIndicator } from 'react-native';
import styled from 'styled-components/native';

interface LoaderProps {
  marginTop: number;
  size: string;
}

const Loader = ({ marginTop, size }: LoaderProps) => (
  <Container>
    <ActivityIndicator
      // @ts-ignore
      size={size}
      color="#eb9e38"
      style={{ marginTop: marginTop }}
    />
  </Container>
);

Loader.defaultProps = {
  size: 'small',
};

const Container = styled.View`
  flex: 1;
  justify-content: center;
`;

export default Loader;
