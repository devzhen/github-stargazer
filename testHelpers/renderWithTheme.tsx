import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';
import 'jest-styled-components';

import theme from '@theme/index';

const renderWithTheme = (component: any) =>
  renderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>);

export default renderWithTheme;
