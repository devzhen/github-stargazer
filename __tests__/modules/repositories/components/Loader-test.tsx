import React from 'react';
import 'jest-styled-components';

import Loader from '@modules/repositories/components/Loader';

// @ts-ignore
import renderWithTheme from '@testHelpers/renderWithTheme';

describe('Loader repository component', () => {
  it('renders correctly, 1 case', () => {
    const tree = renderWithTheme(
      <Loader marginTop={100} size="large" />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly, 2 case', () => {
    const tree = renderWithTheme(
      <Loader marginTop={100} size="small" />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly, 3 case', () => {
    const tree = renderWithTheme(<Loader marginTop={0} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
