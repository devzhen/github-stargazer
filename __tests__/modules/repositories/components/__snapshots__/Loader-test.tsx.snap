// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader repository component renders correctly, 1 case 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#eb9e38"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "marginTop": 100,
      }
    }
  />
</View>
`;

exports[`Loader repository component renders correctly, 2 case 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#eb9e38"
    hidesWhenStopped={true}
    size="small"
    style={
      Object {
        "marginTop": 100,
      }
    }
  />
</View>
`;

exports[`Loader repository component renders correctly, 3 case 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#eb9e38"
    hidesWhenStopped={true}
    size="small"
    style={
      Object {
        "marginTop": 0,
      }
    }
  />
</View>
`;
